File: core/src/com/bogdan/puzzle/level/LevelReader.java
package com.bogdan.puzzle.level;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.files.FileHandle;
import com.google.gson.Gson;
import org.hexworks.mixite.core.api.HexagonOrientation;
import org.hexworks.mixite.core.api.HexagonalGridLayout;

import java.util.ArrayList;

import static com.bogdan.puzzle.level.LevelController.LEVELS_PATH;

class LevelReader {

    private Gson gson;

    LevelReader(){
        gson = new Gson();
    }

    LevelModel loadLevel(int levelId){
        LevelModel levelModel = null;

        if(Gdx.files.internal(LEVELS_PATH + "level_" + levelId + ".json").exists()){
            FileHandle handle = Gdx.files.internal(LEVELS_PATH + "level_" + levelId + ".json");
            // Read all the file in a string
            String levelJson = handle.readString();
            levelModel = gson.fromJson(levelJson, LevelModel.class);
        } else {
            System.out.println("Level not found!");
        }

        return levelModel;
    }

}

File: core/src/com/bogdan/puzzle/level/LevelReader.kt
package com.bogdan.puzzle.level

import com.badlogic.gdx.Gdx
import com.bogdan.puzzle.exceptions.LevelNotFoundException
import com.bogdan.puzzle.level.LevelConstants.LEVELS_PATH
import com.bogdan.puzzle.model.LevelModel
import com.google.gson.Gson

class LevelReader {
    private var gson: Gson = Gson()

    fun loadLevel(levelId: Int): LevelModel {
        val levelModel: LevelModel
        if (Gdx.files.internal(LEVELS_PATH + "level_" + levelId + ".json").exists()) {

            val handle = Gdx.files.internal(LEVELS_PATH + "level_" + levelId + ".json")
            // Read all the file in a string

            val levelString = handle.readString()
            levelModel = gson.fromJson(levelString, LevelModel::class.java)
        } else {
            throw LevelNotFoundException("Level $levelId not found at ${LEVELS_PATH}level_${levelId}.json")
        }
        return levelModel
    }
}